services/src/main/java/org/keycloak/services/resources/admin/AuthenticationManagementResource.java

```diff
@@ -566,6 +566,9 @@ public void recurseExecutions(AuthenticationFlowModel flow, List<AuthenticationE
             } else {
                 String providerId = execution.getAuthenticator();
                 ConfigurableAuthenticatorFactory factory = CredentialHelper.getConfigurableAuthenticatorFactory(session, providerId);
+                if (factory == null) {
+                    throw new NotFoundException("Could not find authenticator provider");
```

Comment:
Minor: will it be possible to add `logger.warnf("Cannot find authentication provider implementation with provider ID '%s'", providerId);`

So that user is aware of what the missing authenticator is?

Reply:
I made this change.

Reply:
Thanks!